### ISSUE TITLE
choco proxy support

### ISSUE THREAD

**Issue by JonCubed** [chocolatey/choco#243]
I'm using chocolatey in automated builds to install some stuff and the http_proxy and https_proxy environment variables do not work in 0.9.9.

The documentation https://github.com/chocolatey/choco/wiki/Proxy-Settings-for-Chocolatey appears to be wrong.

I have solved this by adding 

``` xml
  <system.net>
    <defaultProxy>
      <proxy proxyaddress="<proxyserver>" />
    </defaultProxy>
  </system.net>
```

to _chocolatey.config_ and also updating _Get-WebFile.ps1_ & _Get-WebHeaders.ps1_ in _/helpers/functions_ to setup the proxy even if `$webclient.Proxy.IsBypassed($url)` returns true

Would be nice if the environment variables where supported again.

**Comment by antonyfrancis** [comment]
Yes. Have the same issue. Would be great is environment variables are supported. 

Many thanks.

**Comment by ferventcoder** [comment]
Thanks for creating this with things that can be done to fix this as well!

**Comment by ferventcoder** [comment]
Found more information on this. Has to do with the config needing not to be blank. We'll likely need to override the defaults, which don't find the chocolatey.config and don't find the nuget.config unless it exists (but that is not an optimal way to go about it).

For now a workaround is to create the Nuget.Config file like before with the right settings.

**Comment by ferventcoder** [comment]
This is issue https://github.com/chocolatey/chocolatey/issues/404 from the old repo.

**Comment by youRFate** [comment]
There also seems to be a problem even if the proxy is configured in IE. I have the proxy set up in IE and have it set in the environment variables http_proxy and https_proxy. I installed choco by manually downloading and then running the installer in powershell. But when I try to intstall `ag` for example, I get this error (proxy address edited):

     Using this proxyserver: my.proxy.server
     Using this proxyserver: my.proxy.server
     WARNUNG: Write-ChocolateyFailure is deprecated. If you are the package maintainer, please use 'throw $_.Exception' instead.
     Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (407) Proxy Authentication Required."
     Bei C:\ProgramData\chocolatey\helpers\functions\Write-ChocolateyFailure.ps1:24
    
     Zeichen:8
     +   throw <<<<  "$failureMessage"
         + CategoryInfo          : OperationStopped: (Exception calli...tion Required.":String) [], RuntimeException
         + FullyQualifiedErrorId : Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (407) Proxy Authentication Required."
    The install of ag was NOT successful.
    Error while running 'C:\ProgramData\chocolatey\lib\ag\tools\chocolateyInstall.ps1'.
     See log for details.

The proxy requires username and password authentication.

**Comment by ferventcoder** [comment]
Thanks for the information. I may have something for testing over the next couple of weeks if you are interested. This one is a bit harder for me as I don't have a proxy myself. I know that nuget.core will accept those proxy credentials in the config, which I think we need to ensure the download will also look at those values.

**Comment by youRFate** [comment]
Sure, I can help you test.
Thanks for looking into this.

**Comment by JonCubed** [comment]
Happy to test as well when a fix is available

**Comment by ifunky** [comment]
Hi,

I can help test this when the fix is ready.  The nuget config workaround doesn't seem to work so unfortunately we can't use chocolatey at the moment :-(

Thanks

Dan

**Comment by ferventcoder** [comment]
@ifunky You can use 0.9.8.x series. You may also want to try to set a file next to choco.exe named choco.exe.config with the xml configuration settings in it.

**Comment by ferventcoder** [comment]
Or you can try this (untested):

Create a file at `C:\ProgramData\Chocolatey\choco.exe.config` (based on the default install location, make sure it sits next to `$env:ChocolateyInstall\choco.exe`) with the following contents:

~~~xml
<?xml version="1.0"?>
<configuration>
  <appSettings />
  <system.net>
    <defaultProxy>
      <proxy proxyaddress="<proxyserver>" />
    </defaultProxy>
  </system.net>
</configuration>
~~~

**Comment by ifunky** [comment]
Thanks for the quick response - the choco.exe.config didn't work but putting the proxy settings as above in machine.config did the trick!

Cheers
Dan

**Comment by GuillaumeBonamy** [comment]
@ifunky What do you mean by machine.config?

**Comment by ferventcoder** [comment]
@GuillaumeBonamy the one for .net framework. Only do that if you want it to affect ALL .net applications.

**Comment by ferventcoder** [comment]
This is pretty much complete. Take a look at https://github.com/chocolatey/choco/wiki/Proxy-Settings-for-Chocolatey#explicit-proxy-settings for information on how to set this up once 0.9.9.9 rolls out the door.

**Comment by ferventcoder** [comment]
Merged into stable at https://github.com/chocolatey/choco/commit/bfa8ccfe1ff5202feccfeaa4e28c47ddc892d922 - will be released in 0.9.9.9.

**Comment by JonCubed** [comment]
thanks a lot @ferventcoder
